crates/pallet-subspace/src/lib.rs

```diff
@@ -307,7 +308,7 @@ pub mod pallet {
         #[inline]
         fn default() -> Self {
             Self {
-                enable_rewards_at: EnableRewardsAt::Height(None),
+                enable_rewards_at: EnableRewardsAt::Height(Some(BlockNumberFor::<T>::one())),
```

Comment:
I agree, I will report the issue to `serde_json` with a minimal reproducible example, but IMO it is not a hack for us since `EnableRewardsAt::Height(None)` is used as the same as `EnableRewardsAt::Height(1)` in `genesis_build`

Reply:
I'm pretty confident this is not a bug in `serde_json` or `serde`, it is something on our end. While I'm not opposed to this, I'd prefer to understand what would be the solution first.

Reply:
Thanks @teor2345, I think this is an issue in the substrate, which will remove the kv pair from the json if it is a `null` value so `{"height":null}` will become `{}`:
https://github.com/paritytech/polkadot-sdk/blob/4e2473342fe861687e507a84dcf91e7024a846b9/substrate/client/chain-spec/src/genesis_config_builder.rs#L148-L154
https://github.com/paritytech/polkadot-sdk/blob/4e2473342fe861687e507a84dcf91e7024a846b9/substrate/client/chain-spec/src/json_patch.rs#L36-L38

Reply:
Issue created https://github.com/paritytech/polkadot-sdk/issues/6306